pkg/verify/verify.go

```diff
@@ -128,24 +151,44 @@ func (v *ReleaseVerifier) String() string {
 		}
 		fmt.Fprint(&builder, ")")
 	}
-	fmt.Fprintf(&builder, " - will check for signatures in containers/image format at")
-	if len(v.stores) == 0 {
-		fmt.Fprint(&builder, " <ERROR: no stores>")
-	}
-	for i, store := range v.stores {
-		if i != 0 {
-			fmt.Fprint(&builder, ",")
+
+	hasLocations := len(v.locations) > 0
+	hasStores := len(v.stores) > 0
+	if hasLocations || hasStores {
+		fmt.Fprintf(&builder, " - will check for signatures in containers/image format")
+		if hasLocations {
+			fmt.Fprintf(&builder, " at")
+			for i, location := range v.locations {
+				if i != 0 {
+					fmt.Fprint(&builder, ",")
+				}
+				fmt.Fprintf(&builder, " %s", location.String())
+			}
+		}
+		if hasStores {
+			if hasLocations {
+				fmt.Fprintf(&builder, " and from")
+			} else {
+				fmt.Fprintf(&builder, " from")
+			}
+			for i, store := range v.stores {
+				if i != 0 {
+					fmt.Fprint(&builder, ",")
+				}
+				fmt.Fprintf(&builder, " %s", store.String())
+			}
 		}
-		fmt.Fprintf(&builder, " %s", store.String())
+	} else {
+		fmt.Fprintf(&builder, " - <ERROR: no locations or stores>")
 	}
 	return builder.String()
 }
 
 // Verify ensures that at least one valid signature exists for an image with digest
-// matching release digest in any of the provided stores for all verifiers, or returns
+// matching release digest in any of the provided locations for all verifiers, or returns
```

Comment:
`locations` -> `stores`?  Looks like you're checking both, so probably want to keep `stores` anyhow (and I think we want to [fold `locations` into `stores`][1], which would remove the distinction.

[1]: https://github.com/openshift/cluster-version-operator/pull/279/commits/abd6802c5a4c1fd840b28bb5d516d28431fee228#r359092643